pkg/noderesourcetopology/cache/overallocating_test.go

```diff
@@ -0,0 +1,792 @@
+						},
+					},
+				},
+			},
+		},
+	}
+	nrtCache.ReserveNodeResources("node1", testPod)
+
+	nrtObj := nrtCache.GetByNode("node1", testPod)
+	for _, zone := range nrtObj.Zones {
+		for _, zoneRes := range zone.Resources {
+			switch zoneRes.Name {
+			case string(corev1.ResourceCPU):
+				if zoneRes.Available.Cmp(resource.MustParse("22")) != 0 {
+					t.Errorf("quantity mismatch in zone %q", zoneRes.Name)
+				}
+			case string(corev1.ResourceMemory):
+				if zoneRes.Available.Cmp(resource.MustParse("44Gi")) != 0 {
+					t.Errorf("quantity mismatch in zone %q", zoneRes.Name)
+				}
+			}
+		}
+	}
+}
+
+func TestGetByNodeReleaseNone(t *testing.T) {
+	fakeClient := faketopologyv1alpha1.NewSimpleClientset()
+	fakeInformer := topologyinformers.NewSharedInformerFactory(fakeClient, 0).Topology().V1alpha1().NodeResourceTopologies()
+	fakeIndex := &fakePodByNodeNameIndex{}
+
+	nrtCache, err := NewOverallocating(fakeInformer.Lister(), fakeIndex)
+	if err != nil {
+		t.Fatalf("unexpected error creating cache: %v", err)
+	}
+
+	nodeTopologies := []*topologyv1alpha1.NodeResourceTopology{
+		{
+			ObjectMeta:       metav1.ObjectMeta{Name: "node1"},
+			TopologyPolicies: []string{string(topologyv1alpha1.SingleNUMANodeContainerLevel)},
+			Zones: topologyv1alpha1.ZoneList{
+				{
+					Name: "node-0",
+					Type: "Node",
+					Resources: topologyv1alpha1.ResourceInfoList{
+						MakeTopologyResInfo(cpu, "32", "30"),
+						MakeTopologyResInfo(memory, "64Gi", "60Gi"),
+						MakeTopologyResInfo(nicResourceName, "16", "16"),
+					},
+				},
+				{
+					Name: "node-1",
+					Type: "Node",
+					Resources: topologyv1alpha1.ResourceInfoList{
+						MakeTopologyResInfo(cpu, "32", "30"),
+						MakeTopologyResInfo(memory, "64Gi", "60Gi"),
+						MakeTopologyResInfo(nicResourceName, "16", "16"),
+					},
+				},
+			},
+		},
+	}
+	for _, obj := range nodeTopologies {
+		nrtCache.Store().Update(obj)
+	}
+
+	testPod := &corev1.Pod{
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
+				{
+					Resources: corev1.ResourceRequirements{
+						Limits: corev1.ResourceList{
+							corev1.ResourceCPU:    resource.MustParse("8"),
+							corev1.ResourceMemory: resource.MustParse("16Gi"),
+						},
+						Requests: corev1.ResourceList{
+							corev1.ResourceCPU:    resource.MustParse("8"),
+							corev1.ResourceMemory: resource.MustParse("16Gi"),
+						},
+					},
+				},
+			},
+		},
+	}
+	nrtCache.UnreserveNodeResources("node1", testPod)
+
+	nrtObj := nrtCache.GetByNode("node1", testPod)
+	for _, zone := range nrtObj.Zones {
+		for _, zoneRes := range zone.Resources {
+			switch zoneRes.Name {
+			case string(corev1.ResourceCPU):
+				if zoneRes.Available.Cmp(resource.MustParse("30")) != 0 {
+					t.Errorf("quantity mismatch in zone %q", zoneRes.Name)
+				}
+			case string(corev1.ResourceMemory):
+				if zoneRes.Available.Cmp(resource.MustParse("60Gi")) != 0 {
+					t.Errorf("quantity mismatch in zone %q", zoneRes.Name)
+				}
+			}
+		}
+	}
+}
```

Comment:
I think we should streamline this logic as the NRT object has not changed:
```
if !reflect.DeepEqual(nrtObj, nodeTopologies[0]) {
		t.Fatalf("unexpected object from cache\ngot: %s\nexpected: %s\n", dumpNRT(nrtObj), dumpNRT(nodeTopologies[0]))
	}
```

Reply:
right, done.